### ISSUE TITLE
"/camera/imu" is not published by D435i.

### ISSUE THREAD

**Issue by armanhamzehlou** [realsenseai/realsense-ros#1129]
Hello,

I am following some tutorials on ROS.org:
http://wiki.ros.org/rtabmap_ros/Tutorials/HandHeldMapping
http://wiki.ros.org/rtabmap_ros/Tutorials/RemoteMapping
Both of these tutorials point to a topic called "/camera/imu". However, under my published topics, I do not find it:

```
/camera/accel/imu_info
/camera/accel/sample
/camera/color/camera_info
/camera/color/image_raw
/camera/color/image_raw/compressed
/camera/color/image_raw/compressed/parameter_descriptions
/camera/color/image_raw/compressed/parameter_updates
/camera/color/image_raw/compressedDepth
/camera/color/image_raw/compressedDepth/parameter_descriptions
/camera/color/image_raw/compressedDepth/parameter_updates
/camera/color/image_raw/theora
/camera/color/image_raw/theora/parameter_descriptions
/camera/color/image_raw/theora/parameter_updates
/camera/depth/camera_info
/camera/depth/image_rect_raw
/camera/depth/image_rect_raw/compressed
/camera/depth/image_rect_raw/compressed/parameter_descriptions
/camera/depth/image_rect_raw/compressed/parameter_updates
/camera/depth/image_rect_raw/compressedDepth
/camera/depth/image_rect_raw/compressedDepth/parameter_descriptions
/camera/depth/image_rect_raw/compressedDepth/parameter_updates
/camera/depth/image_rect_raw/theora
/camera/depth/image_rect_raw/theora/parameter_descriptions
/camera/depth/image_rect_raw/theora/parameter_updates
/camera/extrinsics/depth_to_color
/camera/extrinsics/depth_to_infra1
/camera/gyro/imu_info
/camera/gyro/sample
/camera/infra1/camera_info
/camera/infra1/image_rect_raw
/camera/infra1/image_rect_raw/compressed
/camera/infra1/image_rect_raw/compressed/parameter_descriptions
/camera/infra1/image_rect_raw/compressed/parameter_updates
/camera/infra1/image_rect_raw/compressedDepth
/camera/infra1/image_rect_raw/compressedDepth/parameter_descriptions
/camera/infra1/image_rect_raw/compressedDepth/parameter_updates
/camera/infra1/image_rect_raw/theora
/camera/infra1/image_rect_raw/theora/parameter_descriptions
/camera/infra1/image_rect_raw/theora/parameter_updates
/camera/motion_module/parameter_descriptions
/camera/motion_module/parameter_updates
/camera/realsense2_camera_manager/bond
/camera/rgb_camera/auto_exposure_roi/parameter_descriptions
/camera/rgb_camera/auto_exposure_roi/parameter_updates
/camera/rgb_camera/parameter_descriptions
/camera/rgb_camera/parameter_updates
/camera/stereo_module/auto_exposure_roi/parameter_descriptions
/camera/stereo_module/auto_exposure_roi/parameter_updates
/camera/stereo_module/parameter_descriptions
/camera/stereo_module/parameter_updates
/diagnostics
/rosout
/rosout_agg
/tf
/tf_static
```

Please note that I have updated my Camera's firmware to the latest version since a week ago.
I am running it on an ubuntu client and server environment and trying to pass the data from my Robot server to laptop client so rtabmap on the client can visualize it. However, /camera/imu is not found among topics.

I am using the default rs_camera.launch file along with some parameters using the following command on my server: 

`roslaunch realsense2_camera rs_camera.launch fisheye_fps:=15 depth_fps:=15 infra_fps:=15 color_fps:=15`

I have also seen other users able to get this topic published:
see issue #927 
The above user has been able to get it published using the following command:
`rostopic echo /camera/imu`

If it has been removed from the framework, How can I achieve a remote mapping scenario in an alternative way? (which topic or topics)

**Comment by ashirsat** [comment]
Hi,

The imu reading split in two topics and are available in
/camera/accel/sample and /camera/gyro/sample.


Regards,
Aniket Shirsat

**Comment by doronhi** [comment]
As @ashirsat said, the device produces 2 streams: accel and gyro.
If you need them united into the same topic you can use the option `unite_imu_method:=linear_interpolation` or `unite_imu_method:=copy`.
Search "unite_imu_method" for further explanation in the README.md file or [here](https://github.com/IntelRealSense/realsense-ros)

**Comment by armanhamzehlou** [comment]
> As @ashirsat said, the device produces 2 streams: accel and gyro.
> If you need them united into the same topic you can use the option `unite_imu_method:=linear_interpolation` or `unite_imu_method:=copy`.
> Search "unite_imu_method" for further explanation in the README.md file or [here](https://github.com/IntelRealSense/realsense-ros)

Thank you for your quick response, I followed your advice and read the documentation, I was able to get /camera/imu published using the following command on my server (robot):

`roslaunch realsense2_camera rs_camera.launch fisheye_fps:=15 depth_fps:=15 infra_fps:=15 color_fps:=15  align_depth:=true unite_imu_method:="linear_interpolation"`

I then checked the topics on my client:

![image](https://user-images.githubusercontent.com/48771882/76933773-45abff00-68c4-11ea-8470-7718888f873c.png)
![image](https://user-images.githubusercontent.com/48771882/76933742-33ca5c00-68c4-11ea-8908-84682af7dec4.png)

If I use the compressed version it will throw an exception and asks to use image_transport=compressed. So I ran the rtabmap using the following command on my client:

`roslaunch rtabmap_ros rtabmap.launch rtabmap_args:="--delete_db_on_start" depth_topic:=/camera/aligned_depth_to_color/image_raw rgb_topic:=/camera/color/image_raw/ camera_info_topic:=/camera/color/camera_info approx_sync:=true imu_topic:=/camera/imu queue_size:=100 visual_odometry:=false _image_transport:=compressed
`

and this occures:

![image](https://user-images.githubusercontent.com/48771882/76933944-a804ff80-68c4-11ea-99af-1923aa3e314b.png)



- I have already tried toggling approx_sync to false with no luck. increasing queue_size to 1000 doesn't help either. 
- date and times on both machines are identical.
- All launch files in this work are the default ones provided by the original packages.

**Comment by doronhi** [comment]
It seems from the `rostopic hz` command's output that frames are not being published on the following topics: `/camera/aligned_depth_to_color/image_raw`, `/camera/color/image_raw`. Can you confirm?
If so, that seems to be the issue. Are they still not being published using rs_camera.launch file without parameters? What is output? What machine and OS do you use?

**Comment by armanhamzehlou** [comment]
> It seems from the `rostopic hz` command's output that frames are not being published on the following topics: `/camera/aligned_depth_to_color/image_raw`, `/camera/color/image_raw`. Can you confirm?
> If so, that seems to be the issue. Are they still not being published using rs_camera.launch file without parameters? What is the output? What machine and OS do you use?

You are right. They are not being published. As you see in the image in my previous reply, The topics you mentioned publish on a very irregular base. every 5-6 seconds. 

As for vanilla rs_camera.launch I am attaching the logs of the Server and Client and rostopics command.

**### Server Log:**

<pre><font color="#8AE234"><b>arman@arman-Surface-Pro-6</b></font>:<font color="#729FCF"><b>~</b></font>$ roslaunch realsense2_camera rs_camera.launch
... logging to /home/arman/.ros/log/f7608c3a-69ae-11ea-94a9-2816a86909bd/roslaunch-arman-Surface-Pro-6-1840.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is &lt;1GB.

<b>started roslaunch server http://192.168.0.23:44143/</b>

SUMMARY
========

PARAMETERS
 * /camera/realsense2_camera/accel_fps: 250
 * /camera/realsense2_camera/accel_frame_id: camera_accel_frame
 * /camera/realsense2_camera/accel_optical_frame_id: camera_accel_opti...
 * /camera/realsense2_camera/align_depth: False
 * /camera/realsense2_camera/aligned_depth_to_color_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_fisheye1_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_fisheye2_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_fisheye_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_infra1_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_infra2_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/allow_no_texture_points: False
 * /camera/realsense2_camera/base_frame_id: camera_link
 * /camera/realsense2_camera/calib_odom_file: 
 * /camera/realsense2_camera/clip_distance: -2.0
 * /camera/realsense2_camera/color_fps: 30
 * /camera/realsense2_camera/color_frame_id: camera_color_frame
 * /camera/realsense2_camera/color_height: 480
 * /camera/realsense2_camera/color_optical_frame_id: camera_color_opti...
 * /camera/realsense2_camera/color_width: 640
 * /camera/realsense2_camera/depth_fps: 30
 * /camera/realsense2_camera/depth_frame_id: camera_depth_frame
 * /camera/realsense2_camera/depth_height: 480
 * /camera/realsense2_camera/depth_optical_frame_id: camera_depth_opti...
 * /camera/realsense2_camera/depth_width: 640
 * /camera/realsense2_camera/device_type: 
 * /camera/realsense2_camera/enable_accel: True
 * /camera/realsense2_camera/enable_color: True
 * /camera/realsense2_camera/enable_depth: True
 * /camera/realsense2_camera/enable_fisheye1: True
 * /camera/realsense2_camera/enable_fisheye2: True
 * /camera/realsense2_camera/enable_fisheye: True
 * /camera/realsense2_camera/enable_gyro: True
 * /camera/realsense2_camera/enable_infra1: True
 * /camera/realsense2_camera/enable_infra2: True
 * /camera/realsense2_camera/enable_pointcloud: False
 * /camera/realsense2_camera/enable_sync: False
 * /camera/realsense2_camera/filters: 
 * /camera/realsense2_camera/fisheye1_frame_id: camera_fisheye1_f...
 * /camera/realsense2_camera/fisheye1_optical_frame_id: camera_fisheye1_o...
 * /camera/realsense2_camera/fisheye2_frame_id: camera_fisheye2_f...
 * /camera/realsense2_camera/fisheye2_optical_frame_id: camera_fisheye2_o...
 * /camera/realsense2_camera/fisheye_fps: 30
 * /camera/realsense2_camera/fisheye_frame_id: camera_fisheye_frame
 * /camera/realsense2_camera/fisheye_height: 480
 * /camera/realsense2_camera/fisheye_optical_frame_id: camera_fisheye_op...
 * /camera/realsense2_camera/fisheye_width: 640
 * /camera/realsense2_camera/gyro_fps: 400
 * /camera/realsense2_camera/gyro_frame_id: camera_gyro_frame
 * /camera/realsense2_camera/gyro_optical_frame_id: camera_gyro_optic...
 * /camera/realsense2_camera/imu_optical_frame_id: camera_imu_optica...
 * /camera/realsense2_camera/infra1_frame_id: camera_infra1_frame
 * /camera/realsense2_camera/infra1_optical_frame_id: camera_infra1_opt...
 * /camera/realsense2_camera/infra2_frame_id: camera_infra2_frame
 * /camera/realsense2_camera/infra2_optical_frame_id: camera_infra2_opt...
 * /camera/realsense2_camera/infra_fps: 30
 * /camera/realsense2_camera/infra_height: 480
 * /camera/realsense2_camera/infra_width: 640
 * /camera/realsense2_camera/initial_reset: False
 * /camera/realsense2_camera/json_file_path: 
 * /camera/realsense2_camera/linear_accel_cov: 0.01
 * /camera/realsense2_camera/odom_frame_id: camera_odom_frame
* /camera/realsense2_camera/pointcloud_texture_index: 0
 * /camera/realsense2_camera/pointcloud_texture_stream: RS2_STREAM_COLOR
 * /camera/realsense2_camera/pose_frame_id: camera_pose_frame
 * /camera/realsense2_camera/pose_optical_frame_id: camera_pose_optic...
 * /camera/realsense2_camera/publish_odom_tf: True
 * /camera/realsense2_camera/publish_tf: True
 * /camera/realsense2_camera/rosbag_filename: 
 * /camera/realsense2_camera/serial_no: 
 * /camera/realsense2_camera/tf_publish_rate: 0.0
 * /camera/realsense2_camera/topic_odom_in: odom_in
 * /camera/realsense2_camera/unite_imu_method: 
 * /camera/realsense2_camera/usb_port_id: 
 * /rosdistro: melodic
 * /rosversion: 1.14.4

NODES
  /camera/
    realsense2_camera (nodelet/nodelet)
    realsense2_camera_manager (nodelet/nodelet)

auto-starting new master
<b>process[master]: started with pid [1851]</b>
<b>ROS_MASTER_URI=http://localhost:11311</b>

<b>setting /run_id to f7608c3a-69ae-11ea-94a9-2816a86909bd</b>
<b>process[rosout-1]: started with pid [1862]</b>
started core service [/rosout]
<b>process[camera/realsense2_camera_manager-2]: started with pid [1865]</b>
<b>process[camera/realsense2_camera-3]: started with pid [1868]</b>
[ INFO] [1584601079.204286043]: Initializing nodelet with 8 worker threads.
[ INFO] [1584601079.870000814]: RealSense ROS v2.2.13
[ INFO] [1584601079.870098572]: Running with LibRealSense v2.33.1
[ INFO] [1584601079.896996565]:  
 19/03 02:57:59,980 WARNING [140206037133056] (types.cpp:49) Accel Sensitivity:hwmon command 0x4f failed. Error type: No data to return (-21).
 19/03 02:57:59,990 WARNING [140206037133056] (ds5-motion.cpp:463) IMU Calibration is not available, see the previous message
[ INFO] [1584601079.993763184]: Device with serial number 926522071205 was found.

[ INFO] [1584601079.993886413]: Device with physical ID /sys/devices/pci0000:00/0000:00:14.0/usb2/2-1/2-1:1.0/video4linux/video4 was found.
[ INFO] [1584601079.993933321]: Device with name Intel RealSense D435I was found.
[ INFO] [1584601079.995224861]: Device with port number 2-1 was found.
[ INFO] [1584601080.008715261]: getParameters...
[ INFO] [1584601080.077842173]: setupDevice...
[ INFO] [1584601080.077891491]: JSON file is not provided
[ INFO] [1584601080.077924667]: ROS Node Namespace: camera
[ INFO] [1584601080.077949871]: Device Name: Intel RealSense D435I
[ INFO] [1584601080.077972619]: Device Serial No: 926522071205
[ INFO] [1584601080.078017173]: Device physical port: /sys/devices/pci0000:00/0000:00:14.0/usb2/2-1/2-1:1.0/video4linux/video4
[ INFO] [1584601080.078066279]: Device FW version: 05.12.03.00
[ INFO] [1584601080.078095787]: Device Product ID: 0x0B3A
[ INFO] [1584601080.078158121]: Enable PointCloud: Off
[ INFO] [1584601080.078204478]: Align Depth: Off
[ INFO] [1584601080.078242154]: Sync Mode: Off
[ INFO] [1584601080.078395043]: Device Sensors: 
[ INFO] [1584601080.078489716]: Stereo Module was found.
[ INFO] [1584601080.078528152]: RGB Camera was found.
[ INFO] [1584601080.078568247]: Motion Module was found.
[ INFO] [1584601080.078612448]: (Fisheye, 0) sensor isn&apos;t supported by current device! -- Skipping...
[ INFO] [1584601080.078637231]: (Fisheye, 1) sensor isn&apos;t supported by current device! -- Skipping...
[ INFO] [1584601080.078663101]: (Fisheye, 2) sensor isn&apos;t supported by current device! -- Skipping...
[ INFO] [1584601080.078692226]: (Pose, 0) sensor isn&apos;t supported by current device! -- Skipping...
[ INFO] [1584601080.078746151]: num_filters: 0
[ INFO] [1584601080.078771921]: Setting Dynamic reconfig parameters.
[ INFO] [1584601080.214767483]: Done Setting Dynamic reconfig parameters.
[ INFO] [1584601080.236054511]: depth stream is enabled - width: 640, height: 480, fps: 30, Format: Z16
[ INFO] [1584601080.236543329]: infra1 stream is enabled - width: 640, height: 480, fps: 30, Format: Y8
[ INFO] [1584601080.236932980]: infra2 stream is enabled - width: 640, height: 480, fps: 30, Format: Y8
19/03 02:58:00,237 WARNING [140206037133056] (backend-v4l2.cpp:1240) Pixel format 36315752-1a66-a242-9065-d01814a likely requires patch for fourcc code RW16!
[ INFO] [1584601080.261268265]: color stream is enabled - width: 640, height: 480, fps: 30, Format: RGB8
[ INFO] [1584601080.262024283]: setupPublishers...
[ INFO] [1584601080.264736044]: Expected frequency for depth = 30.00000
[ INFO] [1584601080.293939713]: Expected frequency for infra1 = 30.00000
[ INFO] [1584601080.310577567]: Expected frequency for infra2 = 30.00000
[ INFO] [1584601080.325787864]: Expected frequency for color = 30.00000
[ INFO] [1584601080.342577666]: setupStreams...
[ INFO] [1584601080.343462230]: insert Depth to Stereo Module
[ INFO] [1584601080.343503454]: insert Color to RGB Camera
[ INFO] [1584601080.343531464]: insert Infrared to Stereo Module
[ INFO] [1584601080.343568637]: insert Infrared to Stereo Module
[ INFO] [1584601080.343601428]: insert Gyro to Motion Module
[ INFO] [1584601080.343642398]: insert Accel to Motion Module
[ INFO] [1584601080.377397894]: SELECTED BASE:Depth, 0
[ INFO] [1584601080.447382842]: RealSense Node Is Up!
 19/03 02:58:00,487 WARNING [140205739288320] (sensor.cpp:968) HID timestamp not found, switching to Host timestamps.
</pre>

**### Client Log:**

<pre><font color="#8AE234"><b>arman@ubuntu</b></font>:<font color="#729FCF"><b>~</b></font>$ roslaunch rtabmap_ros rtabmap.launch rtabmap_args:=&quot;--delete_db_on_start&quot; depth_topic:=/camera/aligned_depth_to_color/image_raw rgb_topic:=/camera/color/image_raw/ camera_info_topic:=/camera/color/camera_info approx_sync:=true imu_topic:=/camera/imu queue_size:=1000 visual_odometry:=false _image_transport:=compressed
... logging to /home/arman/.ros/log/f7608c3a-69ae-11ea-94a9-2816a86909bd/roslaunch-ubuntu-8169.log
Checking log directory for disk usage. This may take a while.
Press Ctrl-C to interrupt
Done checking log file disk usage. Usage is &lt;1GB.

<b>started roslaunch server http://192.168.0.24:45337/</b>

SUMMARY
========

PARAMETERS
 * /rosdistro: melodic
 * /rosversion: 1.14.4
 * /rtabmap/rtabmap/Mem/IncrementalMemory: true
 * /rtabmap/rtabmap/Mem/InitWMWithAllNodes: false
 * /rtabmap/rtabmap/approx_sync: True
 * /rtabmap/rtabmap/config_path: 
 * /rtabmap/rtabmap/database_path: ~/.ros/rtabmap.db
 * /rtabmap/rtabmap/frame_id: camera_link
 * /rtabmap/rtabmap/ground_truth_base_frame_id: 
 * /rtabmap/rtabmap/ground_truth_frame_id: 
 * /rtabmap/rtabmap/landmark_angular_variance: 9999.0
 * /rtabmap/rtabmap/landmark_linear_variance: 0.0001
 * /rtabmap/rtabmap/map_frame_id: map
 * /rtabmap/rtabmap/odom_frame_id: 
 * /rtabmap/rtabmap/odom_sensor_sync: False
 * /rtabmap/rtabmap/odom_tf_angular_variance: 1.0
 * /rtabmap/rtabmap/odom_tf_linear_variance: 1.0
 * /rtabmap/rtabmap/publish_tf: True
 * /rtabmap/rtabmap/queue_size: 1000
 * /rtabmap/rtabmap/scan_normal_k: 0
 * /rtabmap/rtabmap/subscribe_depth: True
 * /rtabmap/rtabmap/subscribe_rgbd: False
 * /rtabmap/rtabmap/subscribe_scan: False
 * /rtabmap/rtabmap/subscribe_scan_cloud: False
 * /rtabmap/rtabmap/subscribe_stereo: False
 * /rtabmap/rtabmap/subscribe_user_data: False
 * /rtabmap/rtabmap/wait_for_transform_duration: 0.2
 * /rtabmap/rtabmapviz/approx_sync: True
 * /rtabmap/rtabmapviz/frame_id: camera_link
 * /rtabmap/rtabmapviz/odom_frame_id: 
 * /rtabmap/rtabmapviz/queue_size: 1000
 * /rtabmap/rtabmapviz/subscribe_depth: True
 * /rtabmap/rtabmapviz/subscribe_rgbd: False
 * /rtabmap/rtabmapviz/subscribe_scan: False
 * /rtabmap/rtabmapviz/subscribe_scan_cloud: False
 * /rtabmap/rtabmapviz/subscribe_stereo: False
 * /rtabmap/rtabmapviz/wait_for_transform_duration: 0.2

NODES
  /rtabmap/
    rtabmap (rtabmap_ros/rtabmap)
    rtabmapviz (rtabmap_ros/rtabmapviz)

<b>ROS_MASTER_URI=http://192.168.0.23:11311</b>

<b>process[rtabmap/rtabmap-1]: started with pid [8178]</b>
<b>process[rtabmap/rtabmapviz-2]: started with pid [8179]</b>
[ INFO] [1584601111.081797448]: Starting node...
[ INFO] [1584601111.140852481]: Starting node...
[ INFO] [1584601111.399779093]: Initializing nodelet with 8 worker threads.
[ INFO] [1584601111.478431558]: rtabmapviz: Using configuration from &quot;/home/arman/.ros/rtabmap_gui.ini&quot;
[ INFO] [1584601112.701958335]: /rtabmap/rtabmap(maps): map_filter_radius          = 0.000000
[ INFO] [1584601112.701993690]: /rtabmap/rtabmap(maps): map_filter_angle           = 30.000000
[ INFO] [1584601112.702002569]: /rtabmap/rtabmap(maps): map_cleanup                = true
[ INFO] [1584601112.702008844]: /rtabmap/rtabmap(maps): map_always_update          = false
[ INFO] [1584601112.702014143]: /rtabmap/rtabmap(maps): map_empty_ray_tracing      = true
[ INFO] [1584601112.702020798]: /rtabmap/rtabmap(maps): cloud_output_voxelized     = true
[ INFO] [1584601112.702028241]: /rtabmap/rtabmap(maps): cloud_subtract_filtering   = false
[ INFO] [1584601112.702034152]: /rtabmap/rtabmap(maps): cloud_subtract_filtering_min_neighbors = 2
[ INFO] [1584601112.710832436]: /rtabmap/rtabmap(maps): octomap_tree_depth         = 16
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
[ INFO] [1584601113.080892192]: rtabmap: frame_id      = camera_link
[ INFO] [1584601113.080940931]: rtabmap: map_frame_id  = map
[ INFO] [1584601113.080953815]: rtabmap: use_action_for_goal  = false
[ INFO] [1584601113.080963038]: rtabmap: tf_delay      = 0.050000
[ INFO] [1584601113.080986713]: rtabmap: tf_tolerance  = 0.100000
[ INFO] [1584601113.081008704]: rtabmap: odom_sensor_sync   = false
[ INFO] [1584601114.180726401]: Reading parameters from the ROS server...
[ INFO] [1584601117.180382757]: Parameters read = 0
[ INFO] [1584601117.613551795]: /rtabmap/rtabmapviz: subscribe_depth = true
[ INFO] [1584601117.613598631]: /rtabmap/rtabmapviz: subscribe_rgb = false
[ INFO] [1584601117.613607900]: /rtabmap/rtabmapviz: subscribe_stereo = false
[ INFO] [1584601117.613615363]: /rtabmap/rtabmapviz: subscribe_rgbd = false (rgbd_cameras=1)
[ INFO] [1584601117.613634527]: /rtabmap/rtabmapviz: subscribe_odom_info = false
[ INFO] [1584601117.613662738]: /rtabmap/rtabmapviz: subscribe_user_data = false
[ INFO] [1584601117.613673933]: /rtabmap/rtabmapviz: subscribe_scan = false
[ INFO] [1584601117.613702271]: /rtabmap/rtabmapviz: subscribe_scan_cloud = false
[ INFO] [1584601117.613711406]: /rtabmap/rtabmapviz: queue_size    = 1000
[ INFO] [1584601117.613719665]: /rtabmap/rtabmapviz: approx_sync   = true
[ INFO] [1584601117.613733496]: Setup depth callback
[ INFO] [1584601117.633340985]: Setting RTAB-Map parameter &quot;Mem/IncrementalMemory&quot;=&quot;true&quot;
[ INFO] [1584601117.638883660]: Setting RTAB-Map parameter &quot;Mem/InitWMWithAllNodes&quot;=&quot;false&quot;
[ INFO] [1584601117.753713101]: 
/rtabmap/rtabmapviz subscribed to (approx sync):
   /rtabmap/odom,
   /camera/color/image_raw,
   /camera/aligned_depth_to_color/image_raw,
   /camera/color/camera_info
[ INFO] [1584601117.753809434]: rtabmapviz started.
<font color="#C4A000">[ WARN] [1584601122.754384671]: /rtabmap/rtabmapviz: Did not receive data since 5 seconds! Make sure the input topics are published (&quot;$ rostopic hz my_topic&quot;) and the timestamps in their header are set. If topics are coming from different computers, make sure the clocks of the computers are synchronized (&quot;ntpdate&quot;). If topics are not published at the same rate, you could increase &quot;queue_size&quot; parameter (current=1000).</font>
<font color="#C4A000">/rtabmap/rtabmapviz subscribed to (approx sync):</font>
<font color="#C4A000">   /rtabmap/odom,</font>
<font color="#C4A000">   /camera/color/image_raw,</font>
<font color="#C4A000">   /camera/aligned_depth_to_color/image_raw,</font>
<font color="#C4A000">   /camera/color/camera_info</font>
<font color="#C4A000">[ WARN] [1584601127.755559989]: /rtabmap/rtabmapviz: Did not receive data since 5 seconds! Make sure the input topics are published (&quot;$ rostopic hz my_topic&quot;) and the timestamps in their header are set. If topics are coming from different computers, make sure the clocks of the computers are synchronized (&quot;ntpdate&quot;). If topics are not published at the same rate, you could increase &quot;queue_size&quot; parameter (current=1000).</font>
<font color="#C4A000">/rtabmap/rtabmapviz subscribed to (approx sync):</font>
<font color="#C4A000">   /rtabmap/odom,</font>
<font color="#C4A000">   /camera/color/image_raw,</font>
<font color="#C4A000">   /camera/aligned_depth_to_color/image_raw,</font>
<font color="#C4A000">   /camera/color/camera_info</font>
<font color="#C4A000">[ WARN] [1584601132.756149004]: /rtabmap/rtabmapviz: Did not receive data since 5 seconds! Make sure the input topics are published (&quot;$ rostopic hz my_topic&quot;) and the timestamps in their header are set. If topics are coming from different computers, make sure the clocks of the computers are synchronized (&quot;ntpdate&quot;). If topics are not published at the same rate, you could increase &quot;queue_size&quot; parameter (current=1000).</font>
<font color="#C4A000">/rtabmap/rtabmapviz subscribed to (approx sync):</font>
<font color="#C4A000">   /rtabmap/odom,</font>
<font color="#C4A000">   /camera/color/image_raw,</font>
<font color="#C4A000">   /camera/aligned_depth_to_color/image_raw,</font>
<font color="#C4A000">   /camera/color/camera_info</font>
<font color="#C4A000">[ WARN] [1584601137.756512150]: /rtabmap/rtabmapviz: Did not receive data since 5 seconds! Make sure the input topics are published (&quot;$ rostopic hz my_topic&quot;) and the timestamps in their header are set. If topics are coming from different computers, make sure the clocks of the computers are synchronized (&quot;ntpdate&quot;). If topics are not published at the same rate, you could increase &quot;queue_size&quot; parameter (current=1000).</font>
<font color="#C4A000">/rtabmap/rtabmapviz subscribed to (approx sync):</font>
<font color="#C4A000">   /rtabmap/odom,</font>
<font color="#C4A000">   /camera/color/image_raw,</font>
<font color="#C4A000">   /camera/aligned_depth_to_color/image_raw,</font>
<font color="#C4A000">   /camera/color/camera_info</font>
^C[rtabmap/rtabmapviz-2] killing on exit
[rtabmap/rtabmap-1] killing on exit
[ INFO] [1584601141.530007291]: rtabmapviz: ctrl-c catched! Exiting Qt app...
[ INFO] [1584601141.530080168]: rtabmapviz stopping spinner...
<font color="#CC0000">[rtabmap/rtabmap-1] escalating to SIGTERM</font>
shutting down processing monitor...
... shutting down processing monitor complete
<b>done</b>
<font color="#8AE234"><b>arman@ubuntu</b></font>:<font color="#729FCF"><b>~</b></font>$ 
</pre>

 **### Topics listing on the client based on "rs_camera.launch" ran on the server without any parameters**

<pre><font color="#8AE234"><b>arman@ubuntu</b></font>:<font color="#729FCF"><b>~</b></font>$ rostopic hz /camera/aligned_depth_to_color/image_raw /camera/color/image_raw /camera/color/camera_info
subscribed to [/camera/aligned_depth_to_color/image_raw]
subscribed to [/camera/color/image_raw]
subscribed to [/camera/color/camera_info]
          topic              rate   min_delta   max_delta   std_dev   window
============================================================================
/camera/color/camera_info   33.09   9.06e-06    0.05        0.01217   32    

          topic              rate   min_delta   max_delta   std_dev   window
============================================================================
/camera/color/camera_info   31.13   6.199e-06   0.08177     0.02085   59    

          topic              rate   min_delta   max_delta   std_dev   window
============================================================================
/camera/color/camera_info   31.37   6.199e-06   0.1132      0.02372   63    

          topic              rate    min_delta   max_delta   std_dev   window
=============================================================================
/camera/color/image_raw     0.9335   0.494       1.648       0.5772    3     
/camera/color/camera_info   30.4     2.861e-06   1.047       0.09701   3     

          topic              rate   min_delta   max_delta   std_dev   window
============================================================================
/camera/color/image_raw     1.262   0.4504      1.648       0.4964    5     
/camera/color/camera_info   30.63   2.861e-06   1.047       0.08945   5     

          topic             rate   min_delta   max_delta   std_dev   window
===========================================================================
/camera/color/camera_info   30.4   2.861e-06   1.047       0.08904   161   

          topic              rate   min_delta   max_delta   std_dev   window
============================================================================
/camera/color/image_raw     1.1     0.4504      1.835       0.5927    7     
/camera/color/camera_info   30.29   2.861e-06   1.047       0.09192   7     

          topic              rate   min_delta   max_delta   std_dev   window
============================================================================
/camera/color/image_raw     1.254   0.4296      1.835       0.5487    9     
/camera/color/camera_info   29.61   2.861e-06   1.047       0.08842   9     

          topic              rate   min_delta   max_delta   std_dev   window
============================================================================
/camera/color/image_raw     1.348   0.4296      1.835       0.5039    11    
/camera/color/camera_info   30.02   2.861e-06   1.047       0.08423   11    

          topic              rate   min_delta   max_delta   std_dev   window
============================================================================
/camera/color/image_raw     1.424   0.4296      1.835       0.4686    13    
/camera/color/camera_info   29.95   2.861e-06   1.047       0.08116   13    

          topic              rate   min_delta   max_delta   std_dev   window
============================================================================
/camera/color/camera_info   30.05   2.861e-06   1.047       0.08003   322   

          topic              rate   min_delta   max_delta   std_dev   window
============================================================================
/camera/color/camera_info   30.09   2.861e-06   1.096       0.09458   361   

^C          topic              rate   min_delta   max_delta   std_dev   window
============================================================================
/camera/color/image_raw     1.257   0.4296      2.186       0.5822    15    
/camera/color/camera_info   29.98   2.861e-06   1.096       0.09127   15    

</pre>

**As you know the camera/imu is not published without further parameters passed to rs_camera.launch.**

**### System Info:**

**For the server I have tried two different environments:**

- A virtual machine running Ubuntu 18.04 with lastest updates as we speak. 8GB of RAM, Connected to USB 3, 2GB of Video memory and 8 cores processor.
-A surface pro 6 running Ubuntu 18.04 with lastest updates as we speak with 16GB of RAM, SSD, Intel Core i7-8650U, Intel UHD Graphics 620.

**For Client:**

- A Virtual Machine with the same specs of the Virtual Server Machine.

I have reversed the situation to see if there is any change by using the surface pro as a client and the VM as the server and still get the same result.

**These VMs are running on a host with:**

- i9 9900k
- RTX 2080Ti 
- 64GB RAM

**Comment by doronhi** [comment]
I'd rather leave the rtabmap out of the loop for the moment. 
I don't see any related warnings from the realsense2_camera node. Do you get the same results for "rostopic hz" if you run it and the entire ROS environment (ROS master) on the server alone? I wonder if this is a communication issue..

**Comment by armanhamzehlou** [comment]
> I'd rather leave the rtabmap out of the loop for the moment.
> I don't see any related warnings from the realsense2_camera node. Do you get the same results for "rostopic hz" if you run it and the entire ROS environment (ROS master) on the server alone? I wonder if this is a communication issue.

Turns out, You are **absolutely correct**. It does seem like a communication issue. I am doubting my wifi dongle which I have dedicated to the VM.

rostopic hz gave me timely frames on the server not even a single entry missed.

Thank you so much for helping me address the two issues in this topic.

Update : After trying a wired connection everything works perfectly fine on the client. For those curious, I have one note : do not use a USB 2.0 or 2.4 GHz Wi-Fi. USB 3.0 and 5 GHz Wi-Fi will be the only way to go if you don't like wired connection. MAKE SURE your camera is receiving enough power from the USB port. Connect to the USB port closest to the motherboard.
